anvil/src/x11.rs

```diff
@@ -300,9 +275,28 @@ pub fn run_x11(log: Logger) {
                 cursor_visible = true;
             }
 
-            pointer_element.set_position((x as i32, y as i32));
+            let scale = Scale::from(output.current_scale().fractional_scale());
+            let cursor_pos = state.pointer_location;
+            let cursor_pos_scaled = state.pointer_location.to_physical(scale).to_i32_round();
+
+            pointer_element.set_position(cursor_pos.to_i32_round());
             pointer_element.set_status(cursor_guard.clone());
-            custom_space_elements.push(CustomSpaceElements::Pointer(&pointer_element));
+            elements.extend(pointer_element.render_elements(cursor_pos_scaled, scale));
```

Comment:
the pointer element now no longer uses the hotspot, resulting on wrong position of the cursor sometimes